src/backend/x11/mod.rs

```diff
@@ -472,7 +472,7 @@ impl X11Surface {
     }
 
     /// Returns a handle to the GBM device used to allocate buffers.
-    pub fn device(&self) -> &gbm::Device<DrmNode> {
+    pub fn device(&self) -> &Arc<Mutex<gbm::Device<DrmNode>>> {
```

Comment:
If we decide to hide the `Arc<Mutex<_>>` (see comment above), we could return a `MutexGuard<'_, gbm::Device<DrmNode>>` here instead.